src/gdal/ogr_file_format.cpp

```diff
@@ -1408,5 +1416,354 @@ LatLon OgrFileImport::calcAverageLatLon(OGRDataSourceH data_source)
 	return num_coords ? LatLon{ y / num_coords, x / num_coords } : LatLon{};
 }
 
+// ### OgrFileExport ###
+
+OgrFileExport::OgrFileExport(QIODevice *stream, Map *map, MapView *view)
+: Exporter(stream, map, view)
+{
+	// Extract filename from QIODevice
+	output_file = qobject_cast<const QFileDevice*>(stream)->fileName();
+}
+
+void OgrFileExport::doExport()
+{
+	// Choose driver and setup format-specific features
+	QString file_extn = QFileInfo(output_file).suffix();
+	GDALDriverH po_driver;
+
+#ifdef GDAL_DMD_EXTENSIONS
+	// GDAL >= 2.0
+	auto count = GDALGetDriverCount();
+	for (auto i = 0; i < count; ++i)
+	{
+		auto driver_data = GDALGetDriver(i);
+		auto extensions_raw = GDALGetMetadataItem(driver_data, GDAL_DMD_EXTENSIONS, nullptr);
+		auto extensions = QByteArray::fromRawData(extensions_raw, int(qstrlen(extensions_raw)));
+		for (auto pos = 0; pos >= 0; )
+		{
+			auto start = pos ? pos + 1 : 0;
+			pos = extensions.indexOf(' ', start);
+			auto extension = extensions.mid(start, pos - start);
+			if (file_extn == QString::fromLatin1(extension))
+			{
+				auto cap_ogr = GDALGetMetadataItem(driver_data, GDAL_DCAP_VECTOR, nullptr);
+				if (qstrcmp(cap_ogr, "YES") != 0)
+					continue;
+				auto cap_open = GDALGetMetadataItem(driver_data, GDAL_DCAP_CREATE, nullptr);
+				if (qstrcmp(cap_open, "YES") != 0)
+					continue;
+				po_driver = driver_data;
+				break;
+			}
+		}
+	}
+#else
+	const char *psz_driver_name;
+	if (file_extn.compare(QString::fromLatin1("gpx"), Qt::CaseInsensitive))
```

Comment:
`compare()` returns 0 on equality, which is `false` when converted to `bool`.  So you must write:
~~~
if (file_extn.compare(QString::fromLatin1("gpx"), Qt::CaseInsensitive) == 0)